File: digitDigit.js
function squareDigits(num){
    return Number(num.toString().split('').map(x=> x*x).join(''))
  }

console.log(squareDigits(3212))//-->  9414
console.log(squareDigits(2112))//--> 4114
console.log(squareDigits(0))//-->  0

File: digit_digit.py
def square_digits(num):
    lst = list(str(num))
    lst2 = [int(x)*int(x) for x in lst]
    return int("".join(map(str, lst2)))

print(square_digits(3212))#-->  9414
print(square_digits(2112))#--> 4114
print(square_digits(0))#-->  0